### PR TITLE
ios: chat preferences, UI and types

### PR DESCRIPTION
**Pull request by epoberezkin**

### CODE REVIEW

File: apps/ios/Shared/Views/UserSettings/PreferencesView.swift [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+//
+//  Created by Evgeny on 13/11/2022.
+//  Copyright © 2022 SimpleX Chat. All rights reserved.
+//
+
+import SwiftUI
+import SimpleXChat
+
+struct PreferencesView: View {
+    @State var allowFullDeletion = FeatureAllowed.yes
```
Comment (jr-simplex) [inline apps/ios/Shared/Views/UserSettings/PreferencesView.swift:13]:
```suggestion
    @State var allowFullDeletion = FeatureAllowed.no
```

File: apps/ios/Shared/Views/Chat/ContactPreferencesView.swift [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import SwiftUI
+import SimpleXChat
+
+struct ContactPreferencesView: View {
+    @State var allowFullDeletion = ContactFeatureAllowed.yes
+    @State var allowVoice = ContactFeatureAllowed.yes
+    @State var prefs = ContactUserPreferences(
+        fullDelete: ContactUserPreference(
+            enabled: FeatureEnabled(forUser: true, forContact: true),
+            userPreference: .user(preference: Preference(allow: .yes)),
```
Comment (jr-simplex) [inline apps/ios/Shared/Views/Chat/ContactPreferencesView.swift:18]:
```suggestion
            userPreference: .user(preference: Preference(allow: .no)),
```

Comment (jr-simplex) [inline apps/ios/Shared/Views/Chat/ContactPreferencesView.swift:18]:
```suggestion
            enabled: FeatureEnabled(forUser: false, forContact: false),
            userPreference: .user(preference: Preference(allow: .no)),
```

File: apps/ios/Shared/Views/Chat/ContactPreferencesView.swift [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+//
+//  Created by Evgeny on 13/11/2022.
+//  Copyright © 2022 SimpleX Chat. All rights reserved.
+//
+
+import SwiftUI
+import SimpleXChat
+
+struct ContactPreferencesView: View {
+    @State var allowFullDeletion = ContactFeatureAllowed.yes
```
Comment (jr-simplex) [inline apps/ios/Shared/Views/Chat/ContactPreferencesView.swift:13]:
```suggestion
    @State var allowFullDeletion = ContactFeatureAllowed.no
```

File: apps/ios/Shared/Views/Chat/ContactPreferencesView.swift [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            enabled: FeatureEnabled(forUser: true, forContact: true),
+            userPreference: .user(preference: Preference(allow: .yes)),
+            contactPreference: Preference(allow: .no)
+        )
+    )
+
+    var body: some View {
+        VStack {
+            List {
+                featureSection(.fullDelete, .yes, prefs.fullDelete, $allowFullDeletion)
```
Comment (jr-simplex) [inline apps/ios/Shared/Views/Chat/ContactPreferencesView.swift:31]:
```suggestion
                featureSection(.fullDelete, .no, prefs.fullDelete, $allowFullDeletion)
```

File: apps/ios/SimpleXChat/ChatTypes.swift [hunk 5]
```diff
@@ -117,6 +117,181 @@ extension NamedChat {
 
 public typealias ChatId = String
 
+public struct FullPreferences: Decodable {
```
Comment (jr-simplex) [inline apps/ios/SimpleXChat/ChatTypes.swift:120]: maybe it's better to define defaults in types instead of views

Comment (epoberezkin) [reply apps/ios/SimpleXChat/ChatTypes.swift:120]: They were just examples, there should be no defaults in UI, they are set from api responses

**Pull request merged**